Library/ViewModels/DashboardViewModel.swift

```diff
@@ -87,25 +102,65 @@ public protocol DashboardViewModelType {
 public final class DashboardViewModel: DashboardViewModelInputs, DashboardViewModelOutputs,
   DashboardViewModelType {
 
-    public init() {
+  //We use producers ot this properties during view appearance and projects retreaval process 
```

Comment:
Hey @stephencelis so there are some race conditions here in the way projects are refreshed each time the view appears. This is the solution @bormind and I came up with - to use these 'interim' `MutableProperty`s to have a way of clearing the last stored value or basically injecting a `nil` into the stream when necessary.

What we were seeing for example was:

1. Navigate to message thread (`messageThreadNavigatedProperty` emits the `Project` and `MessageThread`).
2. If we navigate away and then back, `projects` will `switchMap` over this again and `self.messageThreadReceived` will emit the same values.

So by doing the following:
```
self.messageThreadReceived <~ Signal.merge(
  self.viewWillDisappearProperty.signal.mapConst(nil),
  self.messageThreadNavigatedProperty.signal
)
```
We can inject a `nil` and ignore it when we need to.

There may certainly be another solution but this one worked for us for now! Happy to hear alternative/better approaches.